File: frontend/app/src/main/java/com/shiyao/ai_story/model/entity/Asset.kt
package com.shiyao.ai_story.model.entity

import androidx.room.Entity
import androidx.room.PrimaryKey


/**
 * 资产实体类
 * 用于在资产库中展示已完成的视频资产
 */
@Entity(
    tableName = "asset"
)
data class Asset(
    @PrimaryKey val id: String,
    val storyId: String,
    val title: String,
    val thumbnailUrl: String? = null,
    val videoUrl: String? = null,
    val duration: Int? = null,
    val status: String, // 状态：generating / completed / failed
    val createdAt: Long?=System.currentTimeMillis()
)

File: backend/model/asset.go
package model

import "time"

type Asset struct {
	ID           uint      `json:"id" gorm:"primaryKey"`
	StoryID      uint      `json:"storyId" gorm:"index"`
	Title        string    `json:"title"`
	VideoURL     string    `json:"videoUrl"`
	ThumbnailURL string    `json:"thumbnailUrl"`
	Duration     int       `json:"duration"`
	Status       int       `json:"status"`
	CreatedAt    time.Time `json:"createdAt"`
}

func (Asset) TableName() string {
	return "assets"
}
